src/main/java/uk/gov/companieshouse/insolvency/data/service/NoopTransactionServiceImpl.java

```diff
@@ -109,7 +110,19 @@ public CompanyInsolvency retrieveCompanyInsolvency(String companyNumber) {
     }
 
     @Override
-    public void deleteInsolvency(String contextId, String companyNumber) {
+    public void deleteInsolvency(String contextId, String companyNumber)
+            throws EmptyResultDataAccessException {
+        Optional<InsolvencyDocument> insolvencyDocumentOptional =
+                insolvencyRepository.findById(companyNumber);
+
+        if (insolvencyDocumentOptional.isEmpty()) {
+            logger.info(String.format(
+                    "Company insolvency doesn't exist for company number %s",
+                    companyNumber));
+            throw new EmptyResultDataAccessException(0);
```

Comment:
It has to throw 404 if there is no record. Throw existing exception IllegalArgumentException.

Reply:
`IllegalArgumentException` will take care of it and return 404 so could fit here indeed!
Perhaps could even remove the logger.info statement and pass it on to the exception but up to you
```
 throw new IllegalArgumentException(String.format(
                        "Company insolvency doesn't exist for company number %s", companyNumber)));
```